components/formats-gpl/src/loci/formats/in/ZeissCZIReader.java

```diff
@@ -617,8 +625,16 @@ protected void initFile(String id) throws FormatException, IOException {
     ms0.sizeT *= phases;
 
     // finish populating the core metadata
-
     int seriesCount = positions * acquisitions * mosaics * angles;
+    //Checkpoint for identifying wrong masterfiles.
+    //Current assumption: Datasets with master files will have seriesCount>1,
+    //This strategy is not the safest way, will need to be reviewed in future.
+    if (seriesCount == 1 && masterFile != null && !wrongMasterFile){
```

Comment:
```
sbesson@c000176:apache_repo $ bftools/showinf -autoscale -nometa 17056/WT\ 0\ \(2\).czi 
Checking file format [Zeiss CZI]
Initializing reader
ZeissCZIReader initializing 17056/WT 0 (2).czi
Checking for master file
Initializing master file 17056/WT 0.czi
Initialization took 0.097s

Reading core metadata
filename = 17056/WT 0 (2).czi
...
````

Similarly to what's been done above, we might want to add a logging statement explaining that we are switching back to the selected file else this is confusing.